src/libraries/System.Runtime/tests/TrimmingTests/AppDomainGetThreadPrincipalTest.cs

```diff
@@ -0,0 +1,15 @@
+using System.Security.Principal;
+using System.Threading;
+
+class Program
+{
+    static int Main(string[] args)
+    {
+        // Thread.CurrentPrincipal calls AppDomain.CurrentDomain.GetThreadPrincipal() which
+        // contains annotation attributes and will require WindowsPrincipal.GetDefaultInstance
+        // and GenericPrincipal.GetDefaultInstance
+        AppDomain.CurrentDomain.SetPrincipalPolicy(PrincipalPolicy.WindowsPrincipal);
+        IPrincipal principal = Thread.CurrentPrincipal;
```

Comment:
Should we verify this is a `WindowsPrincipal`?

Reply:
No need, if the method get's trimmed then the re will be an uncaught exception at runtime for the missing type/member which will return a failing exit code.

Reply:
This could very easily change to return `null` in the future if an exception occurs, and this test wouldn't catch it. I would at least call `principal.GetType()` and check the name is correct.